connect-inject/health_check_resource.go

```diff
@@ -278,7 +279,7 @@ func (h *HealthCheckResource) getConsulClient(pod *corev1.Pod) (*api.Client, err
 	if pod.Annotations[annotationConsulNamespace] != "" {
 		localConfig.Namespace = pod.Annotations[annotationConsulNamespace]
 	}
-	localClient, err := api.NewClient(localConfig)
+	localClient, err := consul.NewClient(localConfig)
```

Comment:
is it a bit weird to be using `localConfig := api.DefaultConfig()` and then passing that to `consul.NewClient(localConfig)` ? 
Without duplicating too much, is it possible to make these calls to the same library?

Reply:
Yeah it's a good idea. Here's what I found:

Out of the 9 places we use `consul.NewClient()`, in 4 of them we call `api.DefaultConfig` so they'd benefit from implementing `DefaultConfig()` in this new package. Note that in all those locations we still end up importing `consul/api` because another part of the code uses it.

In the other 5 locations, we're using the `api.Config` struct directly. I don't think it makes sense to duplicate this struct in our `consul` package because it's got a lot of fields (https://github.com/hashicorp/consul/blob/master/api/api.go#L281) and we'd need to implement a transform function to get it back into the `api.Config` struct.

Here's the commit for what it looks like: https://github.com/hashicorp/consul-k8s/commit/e56248073cd36d4d13cd3b4f09c9bebbbf6be9e8

Do you think we should do it despite not all cases being covered?

Reply:
I'm thinking of the consul package as a place where common use cases for the api are covered there. That makes sense for things like NewClient which is a couple lines of functionality, and very commonly used, but the commit you linked seems like just a redirection for a call to the api package and if we're ok with using the api package in other more custom cases, I don't think we need to redirect one line calls to the consul package.

Reply:
I think we’re good to go on this one, I wanted it more for looks than function and if it doesn’t achieve the looks factor because it cannot be consistently applied then there is no reason to bother with it IMHO 💯

Reply:
okay sounds good thanks for the feedback!